src/BlockDiagonals.jl

```diff
@@ -0,0 +1,241 @@
+module BlockDiagonals
+
+using BlockArrays
+using BlockArrays: AbstractBlockSizes, BlockSizes
+using FillArrays
+using LinearAlgebra
+
+export BlockDiagonal, blocks
+
+"""
+    BlockDiagonal{T, V<:AbstractMatrix{T}} <: AbstractBlockMatrix{T}
+
+A BlockMatrix with square blocks on the diagonal, and zeros off the diagonal.
+"""
+struct BlockDiagonal{T, S<:AbstractBlockSizes} <: AbstractBlockMatrix{T}
+    blocks::Vector{<:AbstractMatrix{T}}
+    block_sizes::S
+    function BlockDiagonal(blocks::Vector{<:AbstractMatrix{T}}, sizes::S) where {T, S<:AbstractBlockSizes}
+        all(is_square, blocks) || throw(ArgumentError("All blocks must be square."))
+        # mapreduce would give Array of Tuples; want Array of Arrays
+        return new{T, S}(blocks, sizes)
+    end
+end
+
+function BlockDiagonal(blocks::AbstractVector{<:AbstractMatrix})
+    # mapreduce would give Array of Tuples; want Array of Arrays
+    bs = BlockSizes([[first(size(b)), last(size(b))] for b in blocks]...)
```

Comment:
```suggestion
    bs = BlockSizes(([first(size(b)), last(size(b))] for b in blocks)...)
```
Non sense making a Array only to splat it,

Reply:
Thanks

`BlockSizes` should really have a nicer constructor... why do i have to make `size(b)::Tuple` into an `Array` anyway?!